tools/run_tests/performance/loadtest_template.py

```diff
@@ -0,0 +1,190 @@
+        server_pool = keys['server_pool']
+        servers = template['spec']['servers']
+        for server in servers:
+            server['pool'] = server_pool
+
+    if keys.get('big_query_table'):
+        template['spec']['big_query_table'] = keys['big_query_table']
+
+    for key in ('timeoutSeconds', 'ttlSeconds'):
+        if keys.get(key):
+            template[key] = keys[key]
+
+
+def loadtest_template(input_file_names: Iterable[str], keys: Mapping[str, str],
+                      metadata: Mapping[str, Any]) -> Dict[str, Any]:
+    """Generates the load test template."""
+    clients = list()
+    servers = list()
+    spec = dict()
+    client_languages = set()
+    server_languages = set()
+    template = {
+        'apiVersion': 'e2etest.grpc.io/v1',
+        'kind': 'LoadTest',
+        'metadata': metadata,
+    }
+    for input_file_name in input_file_names:
+        with open(input_file_name) as f:
+            input_config = yaml.safe_load(f.read())
+
+            if input_config.get('apiVersion') != template['apiVersion']:
+                raise ValueError('Unexpected api version in file {}: {}'.format(
+                    input_file_name, input_config.get('apiVersion')))
+            if input_config.get('kind') != template['kind']:
+                raise ValueError('Unexpected kind in file {}: {}'.format(
+                    input_file_name, input_config.get('kind')))
+
+            for client in input_config['spec']['clients']:
+                if client['language'] in client_languages:
+                    continue
+                clients.append(client)
+                client_languages.add(client['language'])
+
+            for server in input_config['spec']['servers']:
+                if server['language'] in server_languages:
+                    continue
+                servers.append(server)
+                server_languages.add(server['language'])
+
+            input_spec = input_config['spec']
+            del input_spec['clients']
+            del input_spec['servers']
+            del input_spec['scenariosJSON']
+            spec.update(input_config['spec'])
+
+    clients.sort(key=lambda x: x['language'])
+    servers.sort(key=lambda x: x['language'])
+
+    spec.update({
+        'clients': clients,
+        'servers': servers,
+    })
+
+    template['spec'] = spec
+
+    loadtest_set_keys(template, keys)
+
+    return template
+
+
+def main() -> None:
+    argp = argparse.ArgumentParser(
+        description='Creates a load test config generator template.')
+    argp.add_argument('-i',
+                      '--inputs',
+                      action='extend',
+                      nargs='*',
+                      type=str,
+                      required=True,
+                      help='Input files.')
+    argp.add_argument('-o',
+                      '--output',
+                      type=str,
+                      help='Output file. Outputs to stdout if not set.')
+    argp.add_argument('-k',
```

Comment:
the concept of keys and their semantics is a bit odd and it's not explained in the help string, so the values to pass here feels a bit like magic.  Also the name "keys" isn't really descriptive.

Reply:
I also don't understand why some keys have default values and some keys do not. (also, it the default value of the keys being used at all?)

Reply:
Since it looks like there's only a limited subset of supported subst keys (and everything else reports an error), perhaps this would make more sense:
`--inject_client_pool   # inject substitution key ${client_pool}  to set  "spec.clients.client_pool"` (or set client pool directly to a value if the arg has a param set)
`--inject_server_pool    # inject substitution key ${server_pool}  to set  "spec.servers.server_pool"`  
`--inject_timeout_seconds` (inject ${timeout_seconds} if no param is set, or  the value if param is set (`--inject_timeout_seconds 3600`)

that way all the possible values are documented directly in the cmdline help and it's also clearly documented where in the resulting yaml these value will show up.

Reply:
Done. I have made it even simpler than that. There are four options that insert a fixed key in the right places, for instance `${client_pool}` for client pools. That way the user does not need to worry about that when generating the template, and it is always clear which key needs a value for substitution when generating the config.